charts/k8ssandra/templates/cassandra/cassdc.yaml

```diff
@@ -66,9 +66,19 @@ spec:
       credentials_validity_in_ms: {{ .Values.cassandra.auth.cacheValidityPeriodMillis }}
       credentials_update_interval_in_ms: {{ .Values.cassandra.auth.cacheUpdateIntervalMillis }}
     jvm-options:
-      initial_heap_size: "800M"
-      max_heap_size: "800M"
-
+{{- if (index .Values.cassandra.datacenters 0).heap }}
+      initial_heap_size: {{ default "800M" (index .Values.cassandra.datacenters 0).heap.size }}
+      max_heap_size: {{ default "800M" (index .Values.cassandra.datacenters 0).heap.size }}
+      heap_size_young_generation: {{ default "1.6Gi" (index .Values.cassandra.datacenters 0).heap.newGenSize }}
```

Comment:
The young generation is only part of the overall heap, so it cannot be larger than the total heap. It looks like you are setting the young generation to double the total size. My suggestion was to set the young generation to half the total heap size.

Reply:
Do we really want to set these? The options we want to expose differ based on the selected GC.

Reply:
Probably not, for the same reasons we do not want to set container/pod resources. At deployment time, i.e., when `helm install` is run and the manifests are generated we don't know how much memory is available.

Reply:
Yes, the young_generation is incorrect, it should have been the other way around.

Moving the defaults to values.yaml and out of cassdc.yaml.